pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -403,7 +396,15 @@ def barrier(self, *args, **kwargs) -> None:
             torch.distributed.barrier()
 
     def broadcast(self, obj: object, src: int = 0) -> object:
-        return self.dist.broadcast(obj)
+        if not distributed_available():
+            raise RuntimeError(
+                "DDPSpawn is not initialized and torch.distributed is not avalible, can not broadcast object"
+            )
+        obj = [obj]
+        if self.global_rank != 0:
```

Comment:
why global_rank != 0 ? `src` is never used right now :( is this a bug in the existing code?

Reply:
Yeah, it's a bug been pointed out in https://github.com/PyTorchLightning/pytorch-lightning/pull/9414, src was not respected